From ff2ed720cdbc4e552b6474574538f175f94de887 Mon Sep 17 00:00:00 2001
From: Yann Gautier <yann.gautier@foss.st.com>
Date: Wed, 31 Aug 2022 11:26:33 +0200
Subject: [PATCH 072/954] refactor(st-drivers): export struct
 stm32_mce_region_s

The structure stm32_mce_region_s is moved to header file to be usable
in other parts of code.

Upstream-Status: Inappropriate [ST downstream https://github.com/STMicroelectronics/arm-trusted-firmware.git a972b3ecc927b4e9f520c56b6f9199aa8724b757 refactor(st-drivers): export struct stm32_mce_region_s]
Signed-off-by: Yann Gautier <yann.gautier@foss.st.com>
Change-Id: I136ded583f276b39eed7b28d380a574c127552e1
Reviewed-on: https://gerrit.st.com/c/mpu/oe/st/tf-a/+/264425
Reviewed-by: CITOOLS <MDG-smet-aci-reviews@list.st.com>
Reviewed-by: CIBUILD <MDG-smet-aci-builds@list.st.com>
Reviewed-by: Nicolas LE BAYON <nicolas.le.bayon@st.com>
---
 drivers/st/mce/stm32_mce.c     | 10 ----------
 include/drivers/st/stm32_mce.h | 12 +++++++++++-
 2 files changed, 11 insertions(+), 11 deletions(-)

diff --git a/drivers/st/mce/stm32_mce.c b/drivers/st/mce/stm32_mce.c
index fd7bd7ba11..162997c548 100644
--- a/drivers/st/mce/stm32_mce.c
+++ b/drivers/st/mce/stm32_mce.c
@@ -84,16 +84,6 @@ struct mce_dt_id_attr {
 	fdt32_t id_attr[MCE_IP_MAX_REGION_NB];
 };
 
-struct stm32_mce_region_s {
-	uint32_t encrypt_mode;	/*
-				 * Specifies the region encryption mode.
-				 * This parameter can be a value of
-				 * @ref MCE_*_MODE (in driver header file).
-				 */
-	uint32_t start_address;	/* Specifies the region start address */
-	uint32_t end_address;	/* Specifies the region end address */
-};
-
 /*
  * @brief  Configure privileged access to the MCE registers.
  * @param  privilege: Only privileged (true) or all (false) access are granted.
diff --git a/include/drivers/st/stm32_mce.h b/include/drivers/st/stm32_mce.h
index e796ecb316..af94cb44f8 100644
--- a/include/drivers/st/stm32_mce.h
+++ b/include/drivers/st/stm32_mce.h
@@ -1,5 +1,5 @@
 /*
- * Copyright (c) 2020, STMicroelectronics - All Rights Reserved
+ * Copyright (c) 2020-2022, STMicroelectronics - All Rights Reserved
  *
  * SPDX-License-Identifier: BSD-3-Clause
  */
@@ -20,6 +20,16 @@
 /* IP configuration */
 #define MCE_IP_MAX_REGION_NB		1U
 
+struct stm32_mce_region_s {
+	uint32_t encrypt_mode;	/*
+				 * Specifies the region encryption mode.
+				 * This parameter can be a value of
+				 * @ref MCE_*_MODE (in driver header file).
+				 */
+	uint32_t start_address;	/* Specifies the region start address */
+	uint32_t end_address;	/* Specifies the region end address */
+};
+
 void stm32_mce_init(void);
 
 int stm32_mce_write_master_key(uint8_t *mkey);
-- 
2.53.0

